packages/cli/src/bin/pipcook-job.ts

```diff
@@ -1,40 +1,146 @@
-  const err = await get(`${route.job}/stop`, { id });
-  err ? logger.fail(err.message) : logger.success('stop job succeeded');
+  try {
+    await client.job.cancel(id);
+    logger.success('stop job succeeded');
+  } catch (err) {
+    logger.fail(err.message);
+  }
+}
+
+async function log(id: string, opts: any): Promise<void> {
+  const client = initClient(opts.host, opts.port);
+  try {
+    const log = await client.job.log(id);
+    console.log(log);
+  } catch (err) {
+    console.error(err.message);
+    process.exit(1);
+  }
+}
+
+function logCallback(event: string, data: any) {
+  if (event === 'log') {
+    if (data.level === 'info') {
+      logger.info(data.data);
+    } else if (data.level === 'warn') {
+      logger.warn(data.data);
+    }
+  }
+}
+
+async function run(filename: string, opts: any): Promise<JobResp> {
+  const client = initClient(opts.host, opts.port);
+  let config: any;
+  if (!isAbsolute(filename)) {
+    filename = join(process.cwd(), filename);
+  }
+  logger.start(`run pipeline from ${filename}`);
+  try {
+    config = await fs.readJson(filename);
+  } catch (err) {
+    logger.fail(`read pipeline config file error: ${err.message}`);
+  }
+  try {
+    logger.info('start to create pipeline');
+    const pipeline = await client.pipeline.create(config);
+    logger.info(`pipeline is created: ${pipeline.id}, installing`);
+    const installingResp = await client.pipeline.install(pipeline.id, { pyIndex: opts.tuna ? tunaMirrorURI : undefined });
+    await client.pipeline.traceEvent(installingResp.traceId, logCallback);
+    logger.info('pipeline installed successfully, start to run job');
+    const jobRunning = await client.job.run(pipeline.id);
+    logger.info(`job is created: ${jobRunning.id}, running`);
+    await client.pipeline.traceEvent(jobRunning.traceId, logCallback);
+    const job = await client.job.get(jobRunning.id);
+    if (job.status === PipelineStatus.SUCCESS) {
+      logger.info('job is finished successfully');
+      return job;
+    } else if (job.status === PipelineStatus.FAIL) {
+      logger.fail(`job is failed: ${job.error}`);
+    } else {
+      logger.fail(`invalid job status: ${job.status}`);
+    }
+  } catch (err) {
+    logger.fail(`something wrong when running job: ${err.message}`);
+  }
+}
+
+export async function start(filename: string, opts: any): Promise<void> {
+  await run(filename, opts);
 }
 
-async function log(id: string): Promise<void> {
-  const log = await get(`${route.job}/${id}/log`);
-  console.log(log);
+async function runAndDownload (filename: string, opts: any) {
+  const job = await run(filename, opts);
+  const client = initClient(opts.host, opts.port);
+  const outputRootPath = join(process.cwd(), opts.output || 'output');
+  logger.info(`start to download output to ${outputRootPath}`);
+  // remove the output dir
+  await fs.remove(outputRootPath);
+  try {
+    // generate output
+    const fileDownloadResp = await client.job.downloadOutput(job.id);
```

Comment:
Is it possbile to encapsulate logics of compress/uncompress into some public utils

Reply:
Moved to the utils.ts, do you mean we should implement this in `core`?